samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -1384,6 +1391,10 @@ mfxStatus CDecodingPipeline::CreateAllocator()
             MSDK_CHECK_STATUS(sts, "m_hwdev->GetHandle failed");
             sts = m_mfxSession.SetHandle(MFX_HANDLE_VA_DISPLAY, va_dpy);
             MSDK_CHECK_STATUS(sts, "m_mfxSession.SetHandle failed");
+            sts = m_mfxSession.SetFrameAllocator(m_pGeneralAllocator);
```

Comment:
I think this should be moved out of #ifdef LIBVA_SUPPORT and done always for the system memory path.

Reply:
Sure. I will change it. Thanks.